ApiClient/Source/Pages/Queries/SinglePageQuery.swift

```diff
@@ -27,26 +27,32 @@ import Foundation
 import Alamofire
 
 final class SinglePageQuery: Query {
-    typealias Result = JSONPage?
+    typealias Result = JSONPage
     
-    let resource: String
+    let url: URL
+    let host: String = "https://api.schedjoules.com"
     let method: HTTPMethod = .get
     let parameters: Parameters = [:]
     let headers: HTTPHeaders = ["Accept" : "application/json", "Content-Type" : "application/json"]
     
-    required init(resource: String) {
-        self.resource = resource
+    private init(pageID: String, queryItems: [URLQueryItem]) {
+        // Initialize url components from a string
+        var urlComponents = URLComponents(string: "https://api.schedjoules.com/pages/\(pageID)")
```

Comment:
will this automatically urlencode pageID?

Reply:
PageID is numeric, does it require encoding?

Reply:
This ctor takes a string, so the user could pass anything. This ctor should still work when we switch to string IDs.